docs/spec/connection-handshake/L1_2.md

```diff
@@ -0,0 +1,626 @@
+  // Abort, unless this is the first time initializing this connection.
+  abortTransactionUnless(getConnection(local.connectionIdentifier) == nil)
+
+  // Now save this connection end in the local store.
+  // Since this connectionEnd is in state INIT, the relayer will continue with
+  // next steps of the connection handshake protocol.
+  setConnection(local.connectionIdentifier, connectionEnd)
+}
+```
+
+Preconditions:
+  - The connection local end (the input `local` argument) should be valid, i.e., a syntactically correct.
+
+Postconditions:
+  - The module executing this handler stores a `ConnectionEnd` (matching with the input arguments `local` and `remote`) in state `INIT`.
+  - Function produces no output; may abort if preconditions not met.
+  - Uniqueness is satisfied because this handler aborts if a key with the identifier `local.connectionIdentifier` already exists in the store (at-most-once semantics).
+  - To ensure termination of the ICS3 protocol, this handler must execute either on one of the chains or both (at-least-once semantics).
+
+
+###### ConnTryHandler
+
+This is step 2.b.
+
+```golang
+func ConnTryHandler(local ConnectionParty,
+                    remote ConnectionParty,
+                    remoteVersions []String,
+                    proofsHeight uint64,
+                    remoteConnectionProof ConnProof,
+                    remoteClientProof ClientProof)
+{ 
+  // Create local end of the connection.
+  connectionEnd = NewConnectionEnd(
+      'TRYOPEN', local, remote, pickVersion(remoteVersions));
+
+  // Uniqueness check.
+  // Verify that either there is no connection OR it is in INIT state (i.e., 
+  // the Init handler previously executed with these same parameters).
+  currnt = getConnection(local.connectionIdentifier)
+  if currnt != nil {
+    abortTransactionUnless(currnt.state = 'INIT')
+  }
+  abortTransactionUnless(currnt == nil || matchingEnds(connectionEnd, currnt))
+
+  expectedConn = NewConnectionEnd('INIT', remote, local)
+  expectedClient = consensusState(remoteClientProof.height)
+
+  // Proofs generic verification.
+  // Generic proof check: The local client should already be updated with the
+  // height which the two proofs target (proofsHeight).
+  localClientState = clientConsensusState(local.clientIdentifier, proofsHeight)
+  abortTransactionUnless(localClientState != nil)
+
+  // Connection proof specific verification. See `verifyProof`.
+  abortTransactionUnless(verifyProof(
+      localClientState, remoteConnectionProof,
+      connectionPath(local.connectionIdentifier), expectedConn))
+
+  // Client proof verification. Helps ensure state consistency: The remote 
+  // client should be consistent with the state of local party.
+  abortTransactionUnless(remoteClientProof.height <= getCurrentHeight())
+  abortTransactionUnless(verifyProof(localClientState, remoteClientProof,
+        clientPath(local.clientIdentifier, remoteClientProof.height),
+        expectedClient))
```

Comment:
to be consistently aligned with the above `ferifyProof`. I am just going over some but we should try to have nice alignment overall.
```suggestion
  abortTransactionUnless(
        verifyProof(localClientState, remoteClientProof,
        clientPath(local.clientIdentifier, remoteClientProof.height),
        expectedClient))
```